### DISCUSSION TITLE
Publish Gitbook changelog to public

### DISCUSSION THREAD

**Discussion by MattIsenbarger** [GitbookIO/community#350] [0. General bugs and questions] [answered]
Is there an automated way to expose my Gitbook changelog or list a "version number" on my public facing docs?

**Comment by aikoii** [comment, marked as answer]
Hi @MattIsenbarger 👋! While GitBook doesn't offer an automated way to expose a changelog or version number on your public docs, you can achieve versioning by creating different variants. Each variant represents a specific version of your documentation. Here is a link with more info https://docs.gitbook.com/publishing/share/collection-publishing

**Reply by Ifycode** [reply]
This Question and response are sooooooooooooo helpful. Thank you both.
I couldn't understand where to find doc versioning from the documentation.

**Comment by Ifycode** [comment]
Hi again @aikoii @MattIsenbarger 
I'm having an issue with variants.

#

The forest admin example was used in the gitbook docs. 
- Click on **_v7 (v6 Rails)_** from the variant dropdown, and then go to the premium features page: https://docs.forestadmin.com/documentation/v/v7-v6-rails/reference-guide/premium-features
- While still on the v7's premium features page, try to access another variant through the dropdown. This page does not exist in other variants, so it redirects to a page that exists in the variants. Therefore, the other variants are able to open up successfully and without errors (i.e. it goes to the how it works page of those variants since they don't have premium features page).

#

In my own case, it doesn't redirect. It just tries to look for the same page and url for the variant page doc I'm coming from in the variant doc I'm going to. And because a page does not exist, it brings up the `page not found` error page. I have to always click on the variant a second time after the error page occurs to get an existing variant page to successfully show up.

I've looked through the gitbook docs but can't seem to locate anything about how to go about such redirects for variants. How were they able to achieve that? Can you help direct me?

**Reply by Ifycode** [reply]
Reference discussion: https://github.com/orgs/GitbookIO/discussions/375

**Reply by aikoii** [reply]
Hey @Ifycode, I must admit, I'm also curious about why this works for the Forest Admin docs 🤔. I'll sync with our team on that.

So, the design for collections expects page URLs to be consistent across spaces, except for the variant slug. Currently, there's no mechanism to map a specific page in one variant to another in a different variant. Therefore, if there's a mismatch in paths, it could trigger "page not found" errors.

We've talked about this internally and recognize that there is room for improvement! However, just to set expectations, it's not something we're prioritizing immediately.

A workaround you could consider for now is choosing one variant as your "base", as it were, and edit all page/page group URLs in the other variant(s) to match that "base" variant.

**Reply by Ifycode** [reply]
Ok @aikoii I've read it, thank you for your response. I'm still hoping your team will be able to give me the real solution. I have a base, and I will try out what you have asked me to do for now. Just that this means if anything changes in the docs, we would always have to remember to go change the URL... Which is stressful.

**Reply by Ifycode** [reply]
Plus I'm wondering if this won't be confusing for users of the docs because of the nature of our docs. Editing the url to match the base variant even if the content is unrelated to the slug?

**Reply by Ifycode** [reply]
@aikoii There is also the problem where one variant only contains 2 pages, and there are or will be numerous pages for the others. How do I now map few pages to numerous pages?

**Reply by aikoii** [reply]
I completely understand where you're coming from! The current solution isn't perfect and might not always give the desired results 🙏. When mapping a few pages to multiple pages, you can still keep the base URL structure and I think it should work in most cases when switching between variants.
